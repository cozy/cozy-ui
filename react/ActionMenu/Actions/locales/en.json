{
  "viewInContacts": "View in Cozy Contacts",
  "modify": "Modify",
  "emailTo": "Send an email",
  "smsTo": "Send a message",
  "call": "Call"
}
