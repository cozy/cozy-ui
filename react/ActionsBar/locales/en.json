{
  "selected": "%{smart_count} item selected |||| %{smart_count} items selected"
}
