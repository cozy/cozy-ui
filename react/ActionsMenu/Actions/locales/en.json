{
  "viewInContacts": "View in Cozy Contacts",
  "viewInDrive": "Open",
  "modify": "Modify",
  "emailTo": "Send an email",
  "smsTo": "Send a message",
  "print": "Print",
  "others": "Others",
  "call": "Call"
}
