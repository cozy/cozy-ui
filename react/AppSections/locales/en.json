{
  "app_categories": {
    "all": "All categories",
    "banking": "Banking",
    "cozy": "The essentials",
    "education": "Education",
    "energy": "Energy",
    "health": "Health",
    "host_provider": "Host",
    "insurance": "Insurance",
    "isp": "ISP",
    "konnectors": "Services",
    "list_separator": ", ",
    "mes_infos": "MesInfos experimentation",
    "online_services": "Online services",
    "others": "Others",
    "partners": "Partners",
    "press": "Press",
    "productivity": "Productivity",
    "ptnb": "PTNB experimentation",
    "public_service": "Public services",
    "shopping": "Shopping",
    "social": "Social",
    "telecom": "Telecom",
    "transport": "Transportation"
  },
  "sections": {
    "applications": "Applications",
    "konnectors": "Services"
  }
}
