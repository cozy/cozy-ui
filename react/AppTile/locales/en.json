{
  "app_item": {
    "by": "By",
    "installed": "Installed",
    "maintenance": "In maintenance",
    "update": "Update available",
    "favorite": "Added to home page"
  }
}
