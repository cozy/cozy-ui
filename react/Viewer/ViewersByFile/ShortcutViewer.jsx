import get from 'lodash/get'
import React from 'react'

import { useClient, useFetchShortcut } from 'cozy-client'

import OpenwithIcon from '../../Icons/Openwith'
import { ButtonLink } from '../../deprecated/Button'
import { FileDoctype } from '../../proptypes'
import NoViewer from '../NoViewer'
import { withViewerLocales } from '../hoc/withViewerLocales'

const ShortcutViewer = ({ t, file }) => {
  const client = useClient()
  const { shortcutInfos } = useFetchShortcut(client, file.id)
  let url = ''
  if (shortcutInfos) {
    url = new URL(get(shortcutInfos, 'data.attributes.url'))
  }
  return (
    <NoViewer
      file={file}
      renderFallbackExtraContent={() => (
        <ButtonLink
          label={`${t('Viewer.goto', { url: get(url, 'origin', '') })}`}
          icon={OpenwithIcon}
          href={`${get(url, 'origin', '')}`}
          target="_blank"
        />
      )}
    />
  )
}

ShortcutViewer.propTypes = {
  file: FileDoctype.isRequired
}

export default withViewerLocales(ShortcutViewer)
