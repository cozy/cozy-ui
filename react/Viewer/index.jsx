import ViewerContainer from './ViewerContainer'

export { default as Viewer } from './Viewer'
export { default as ViewerContainer } from './ViewerContainer'
export { toolbarPropsPropType } from './proptypes'
export {
  default as ViewerWithCustomPanelAndFooter
} from './ViewerWithCustomPanelAndFooter'

export default ViewerContainer
