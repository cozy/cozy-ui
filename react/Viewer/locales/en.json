{
  "Viewer": {
    "actions": {
      "download": "Download",
      "forward": "Forward"
    },
    "alert": {
      "preparing": "Preparing your filesâ€¦"
    },
    "close": "close",
    "download": "Download",
    "error": {
      "downloadFile": {
        "offline": "You should be connected to download this file"
      },
      "generic": "An error occurred when opening this file, please try again.",
      "missing": "This file is missing",
      "network": "This file could not be loaded. Do you have a working internet connection right now?",
      "noapp": "No application on your device can open this file with mime type: %{fileMime} ",
      "offline": "You should be connected to open this file"
    },
    "goto": "Go to %{url}",
    "next": "Next",
    "openWith": "Open with...",
    "openInOnlyOffice": "Open with Only Office",
    "panel": {
      "certifications": {
        "carbonCopy": {
          "title": "Carbon Copy",
          "caption": "Indicates whether the document is defined as \"authentic and original\" by Cozy Cloud, the host of your Cozy, as it can claim that it comes directly from a third-party service, without having undergone any modification."
        },
        "electronicSafe": {
          "title": "Electronic Safe",
          "caption": "Indicates whether the original document is secured by your personal digital safe with the certifications that give it probative value and a 50-year retention guarantee beyond its deposit."
        }
      },
      "qualification": {
        "date": {
          "title": {
            "addedOn": "Added on",
            "obtentionDate": "Delivered on",
            "issueDate": "Delivered on",
            "expirationDate": "Expiration date",
            "referencedDate": "Referenced date",
            "shootingDate": "Shooting date"
          }
        },
        "identity": "Identity",
        "label": {
          "title": "Label",
          "front": "Front side",
          "back": "Back side"
        },
        "qualification": "Qualification"
      },
      "title": "Useful information"
    },
    "previous": "Previous",
    "retry": "Retry",
    "scaledown": "Zoom out",
    "scaleup": "Zoom in",
    "share": {
      "title": "Link to my document %{name}",
      "text": "Here is a link to my document %{name}: ",
      "error": "Problem with link recovery: %{error}",
      "success": "Your file has been shared with success"
    }
  }
}
