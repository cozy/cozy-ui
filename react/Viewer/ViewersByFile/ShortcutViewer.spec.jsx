import React from 'react'

import CozyClient, { CozyProvider } from 'cozy-client'
import { render, waitFor } from '@testing-library/react'
import I18n from '../../providers/I18n'

import en from '../locales/en.json'

import ShortcutViewer from './ShortcutViewer'

export const locales = {
  en
}

const setup = () => {
  const client = new CozyClient({})
  return (
    <CozyProvider client={client}>
      <I18n lang="en" dictRequire={() => locales['en']}>
        <ShortcutViewer file={{ id: '1' }} />
      </I18n>
    </CozyProvider>
  )
}

describe('Shortcutviewer', () => {
  it('renders the component', async () => {
    const { container } = render(setup())
    await waitFor(() => {
      expect(container).toMatchSnapshot()
    })
  })
})
