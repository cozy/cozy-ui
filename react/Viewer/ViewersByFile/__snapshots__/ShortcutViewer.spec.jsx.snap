// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shortcutviewer renders the component 1`] = `
<div>
  <div
    class="styles__viewer-noviewer___auG-6"
  >
    <svg
      class="styles__icon___23x3R"
      height="140"
      viewBox="0 0 32 32"
      width="160"
    >
      <g
        fill="none"
        fill-rule="evenodd"
      >
        <path
          d="M3 2.002C3 .896 3.89 0 4.997 0H22l7 7v22.996A2 2 0 0127.003 32H4.997A1.995 1.995 0 013 29.998V2.002z"
          fill="#D1D5DB"
        />
        <path
          d="M21.5 0c-.276 0-.5.23-.5.5V8h7.5c.276 0 .5-.232.5-.5V7l-7-7h-.5z"
          fill="#A3ACB8"
        />
      </g>
    </svg>
    <p
      class="styles__viewer-filename___3jZCt"
    />
    <a
      class="styles__c-btn___3kXsk"
      href=""
      target="_blank"
    >
      <span>
        <svg
          aria-hidden="true"
          class="styles__icon___23x3R"
          focusable="false"
          height="16"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M9 0v2h3.5L6 8.5 7.5 10 14 3.5V7h2V1.003A.996.996 0 0014.997 0H9zM7 2V0H1.003A1 1 0 000 1v14c0 .552.445 1 1 1h14c.552 0 1-.438 1-1.003V9h-2v5H2V2h5z"
            fill-rule="evenodd"
          />
        </svg>
        <span>
          Go to 
        </span>
      </span>
    </a>
  </div>
</div>
`;
